pkg/recipes/terraform/install.go

```diff
@@ -78,5 +81,40 @@ func Install(ctx context.Context, installer *install.Installer, tfDir string) (s
 
 	logger.Info(fmt.Sprintf("Terraform latest version installed to: %q", execPath))
 
-	return execPath, nil
+	// Create a new instance of tfexec.Terraform with current Terraform installation path
+	tf, err := NewTerraform(ctx, tfDir, execPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Verify Terraform installation is complete before proceeding
+	for attempt := 0; attempt <= installRetryCount; attempt++ {
+		_, _, err = tf.Version(ctx, false)
+		if err != nil {
+			if attempt < installRetryCount {
+				logger.Info(fmt.Sprintf("Failed to verify Terraform installation: %s. Retrying after %d seconds", err.Error(), retryDelaySecs))
+				metrics.DefaultRecipeEngineMetrics.RecordTerraformInstallVerificationDuration(ctx, installStartTime,
```

Comment:
Adding metrics so we can track how long installation/verification takes in these edge cases. We can adjust the retry count and duration if needed

Reply:
This may create a few metrics for just one installation. Is that the goal?

Reply:
I think it'd be good to track how many times we end up needing to retry and what was the duration. Since this is a flaky test we can't really repro to get the right number of retries and duration between each so this will help us get a gauge. Do you think there's a better place to put the metrics to track that info?